releasenotes/notes/vf2_custom_score_analysis-abb191d56c0c1578.yaml

```diff
@@ -0,0 +1,19 @@
+---
+features:
+  - |
+    Added the ability for analysis passes to set custom heuristic weights
+    for the the :class:`~.VF2Layout` and :class:`~.VF2PostLayout` transpiler
+    passes. If an analysis pass sets the ``vf2_avg_error_map`` key in the
+    property set that will be used for the error weights instead of using
+    the error rates in the backend's :class:`~.Target` (or
+    :class:`~.BackendProperties` for :class:`~.BackendV1`). The value should be
+    an 2d Numpy array where each value represents the avg error rate for that
+    1 or 2 qubit operation on those qubits. ``NaN`` can be used to represent
+    the lackof error or support for running gates on that qubit. For example::
```

Comment:
Having looked further at the implementation now, I guess `np.inf` and `np.nan` would both do exactly the same thing here for the "not supported case", and we'd never trap because it's just a sentinel value that's used in a filter.  The alternative I suppose would be to have `1.0` represent "not possible" and to avoid the `NaN` handling entirely, but that starts to feel even iffier.